lib/bindings/HTTPBindings.js

```diff
@@ -165,8 +165,25 @@ function handleIncomingMeasure(req, res, next) {
         }
     }
 
-    function extractServiceAndSubservice(group, callback) {
-        callback(null, group.service, group.subservice);
+    function findOrCreate(group, callback) {
+        iotAgentLib.getDevice(req.deviceId, group.service, group.subservice, function(error, device) {
+            if (!error && device) {
+                callback(null, device);
+            } else if (error.name === 'DEVICE_NOT_FOUND') {
```

Comment:
That shouldn't be possible, AFAIK, as no error means the function completed successfully, and that means it would have returned a value.